src/awsexplorer/commands/copyArn.ts

```diff
@@ -3,24 +3,39 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import * as vscode from 'vscode'
 import * as nls from 'vscode-nls'
+import { getLogger } from '../../shared/logger'
 const localize = nls.loadMessageBundle()
 
 import { AWSResourceNode } from '../../shared/treeview/nodes/awsResourceNode'
+import { Env, DefaultEnv } from '../../shared/vscode/env'
+import { Window, DefaultWindow } from '../../shared/vscode/window'
 
 const COPY_ARN_DISPLAY_TIMEOUT_MS = 2000
 
-export async function copyArnCommand(node: AWSResourceNode) {
+/**
+ * Copies the arn of the resource represented by the given node.
+ */
+export async function copyArnCommand(
+    node: AWSResourceNode,
+    window: Window = new DefaultWindow(),
+    env: Env = new DefaultEnv()
+): Promise<void> {
     try {
-        await vscode.env.clipboard.writeText(node.getArn())
-        vscode.window.setStatusBarMessage(
+        getLogger().debug(`CopyArn called for ${node}`)
```

Comment:
I implemented `toString` on all of the new Nodes. Think I should add the same for other Nodes? It's quite nice to be able to log useful and generic information about the tree nodes for these shared commands.

Reply:
Changed `log.info(blah)` to `log.info('%O', blah)` per the logging change. Everything will now print the full object by default rather than the `[Object object]` form.